omod/src/main/java/org/openmrs/module/appointmentscheduling/rest/resource/openmrs1_9/AppointmentResource1_9.java

```diff
@@ -127,7 +127,8 @@ public Appointment save(Appointment appointment) {
 	protected Appointment save(Appointment appointment, Boolean allowOverbook) {
 		if (appointment.getId() != null) {
 			// existing appointments get updated
-			return Context.getService(AppointmentService.class).saveAppointment(appointment);
+			Context.getService(AppointmentService.class).changeAppointmentStatus(appointment, appointment.getStatus());
+			return appointment;
 		} else {

```

Comment:
This PR pretty much looks good @jecihjoy ... the one concern I have here is that what happens if you save an appointment and the status hasn't changed?  don't you need to do a check here and only call "saveAppointment" if the status hasn't changed?  Otherwise won't you get a new row in the appointment status table each time you change the status, even if it hasn't changed?

Reply:
I understand that, however the challenge i'm having is that the **appointment** argument in the save method is already an updated object and hence I cant get the previous status of the appointment from it, also doing getById(**appointment.getId**) or getByUuid(**appointment.getUuid**) still returns the updated appointment object FYI the appointment object is updated but its records is not in the database

so i'm introducing this helper method, getMostRecentAppointmentStatusHistory

Reply:
Got it... looks like you made a tweak to this since by my comment, to check the status against the history?   That should do it...